File: app/schemas/recommendation.py
from datetime import datetime

from pydantic import BaseModel


class RecommendationRequest(BaseModel):
    staffing_requirement_id: str


class RecommendationItem(BaseModel):
    employee_id: str
    rank: int
    score: float
    eligibility_status: str
    matched_skills: list[str]
    reason: str


class RecommendationResponse(BaseModel):
    staffing_requirement_id: str
    recommendations: list[RecommendationItem]
    generated_at: datetime
    message: str | None = None

File: frontend/src/types/recommendation.ts
export interface RecommendationRequest {
  staffing_requirement_id: string
}

export interface RecommendationItem {
  employee_id: string
  rank: number
  score: number
  eligibility_status: string
  matched_skills: string[]
  reason: string
}

export interface RecommendationResponse {
  staffing_requirement_id: string
  recommendations: RecommendationItem[]
  generated_at: string
  message: string | null
}